### ISSUE TITLE
Method decorator incorrectly marked as no-member

### ISSUE THREAD

**Issue by atollk** [PyCQA/pylint#3476]
### Steps to reproduce

```
import types

def decorator(target=None):
    return Decorated(target)

class Decorated:
    def __init__(self, target):
        self.msg = 'before'
        self.target = target

    def __call__(self, *args, **kwargs):
        print(self.msg)
        self.target(*args, **kwargs)

    def __get__(self, instance, owner):
        if instance is None:
            return self
        else:
            return types.MethodType(self, instance)

    def set_msg(self, x):
        self.msg = x

class A:
    @decorator
    def foo(self):
        pass

a = A()
a.foo()
a.foo.set_msg('after')
a.foo()
```

### Current behavior

> line 31:0: E1101: Method 'foo' has no 'set_msg' member (no-member)

### Expected behavior

No error message. The code runs fine and outputs as expected

> before
> after

### pylint --version output

> pylint 2.5.0-dev1
> astroid 2.4.0
> Python 3.8.1 (tags/v3.8.1:1b293b6, Dec 18 2019, 22:39:24) [MSC v.1916 32 bit (Intel)]

**Comment by PCManticore** [comment]
Thanks for opening an issue. I'm afraid there's not much we can do here since pylint's inference is rather limited. For reference, you are using decorators, factories and descriptors which is entirely overkill for a static analysis tool like pylint.
